### PR TITLE
Extended the commands that supports sort clause to also support vector search

### LINKED ISSUE #478: Extend JSON API vector search functionality support
Extend vector search functionality for commands that supports sort.

- FindOneAndUpdate
- FindOneAndDelete
- FindOneAndReplace
- UpdateOne
- DeleteOne

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Extended the commands that supports sort clause to also support vector search

**Which issue(s) this PR fixes**:
Fixes #478 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 1]
```diff
@@ -748,4 +748,194 @@ public void errorOperationForVector() {
...
+
+    @Test
+    @Order(2)
+    public void findOneAndUpdate() {
+      String json =
+          """
+        {
+          "findOneAndUpdate": {
+            "sort" : {"$vector" : [0.15, 0.1, 0.1, 0.35, 0.55]},
+            "update" : {"$set" : {"status" : "active"}},
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:770]: Do we also need to test `findOneAndUpdate()` with `$vector` in projection and `$vector` sort?

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 2]
```diff
@@ -748,4 +748,194 @@ public void errorOperationForVector() {
...
+          .body("data.document.new_col", is("new_val"));
+    }
+
+    @Test
+    @Order(3)
+    public void findOneAndReplace() {
+      String json =
+          """
+        {
+          "findOneAndReplace": {
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:841]: Is it worthwhile to test `findOneAndReplace()` with no `$vector` in `replacement`?

**Review by vkarpov15 — COMMENTED** [review]
LGTM so far, just some testing questions

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java (hunk 1 above)
Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:770]: PR for projection handles projection for $vector. It uses findOneAndUpdate command for the testig.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 3]
```diff
@@ -748,4 +748,194 @@ public void errorOperationForVector() {
...
+
+    @Test
+    @Order(4)
+    public void findOneAndDelete() {
+      String json =
+          """
+        {
+          "findOneAndDelete": {
+            "sort" : {"$vector" : [0.15, 0.1, 0.1, 0.35, 0.55]},
+            "replacement" : {"_id" : "3", "username": "user3", "status" : false, "$vector" : [0.12, 0.05, 0.08, 0.32, 0.6]},
```
[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:873]: Fixed it.

**Review by tatu-at-datastax — APPROVED** [review]

**Pull request merged**